dasharo-compatibility/power-after-fail.robot

```diff
@@ -0,0 +1,94 @@
+*** Settings ***
+Library             Collections
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+Resource            ../lib/bios/menus.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keyword
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+PSF001.001 Check Power State After Power Failure default state (firmware)
+    [Documentation]    This test ensures that the option is present, and the
+    ...    default state of this option after flashing is correct.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    PSF001.001 not supported
+    Power On
+    Enter Dasharo System Features
+    Enter Power Management Options
+    ${psf_state}=    Read From Terminal Until    ESC to exit.
+    Should Contain    ${psf_state}    Powered Off
```

Comment:
The option default is firmware build dependent. You would have to parse the coreboot config file to check what should be the default state:

CONFIG_MAINBOARD_POWER_FAILURE_STATE

0 - is power off
1 - is power on
2 - is keep previous state (aka. `The state at the moment of power failure`)

Reply:
I think the simplest option would be adding that as a setting for the compatibility module to the mainboard config? IMO we don't know if we'll be able to readily access the build config file from the position of the test file

Reply:
I gave a generic method that will always work with any binary and it is the most recommended option. Let;s not hardcode things which do not have to be hardcoded.

Reply:
@miczyg1 @filipleple I am fine with defining this default in platform config, just as we do for the default Secure Boot state as far as I remember.

It is true that we could extract such things from coreboot config and expect based on that, but I'm not totally convinced on this approach.

One of the goal of testing is to also define an independent set of expected outcomes and verify that the firmware product delivers that. If we define in the platform config that platform X is expected to ship with feature Y enabled by default, we can check if the firmware developer used correct configuration as well.

Reply:
So, config [option](https://github.com/Dasharo/open-source-firmware-validation/pull/29/files#diff-924a28816186a815414d0e7ac57b47fab4603000ee13fbd0adb8a4952e0b7b3dR160) added and checked for [in the test case](https://github.com/Dasharo/open-source-firmware-validation/pull/29/files#diff-b9e98ddabefa0c5dd70c8174e598bd616df7ae602633134a00d0debe605e2398R38)

Reply:
@filipleple It shall be added to all MSI boards, at least.

Reply:
@macpijan added to all platform configs

Reply:
> One of the goal of testing is to also define an independent set of expected outcomes and verify that the firmware product delivers that. If we define in the platform config that platform X is expected to ship with feature Y enabled by default, we can check if the firmware developer used correct configuration as well.

Then why not have both? The hardcoded EXPECTED state of feature and then the feature state in the binary. Then we could also detect misconfiguration of the binary. Maybe even do a preliminary test to check feature state before starting the real tests

Reply:
Issue for this enhancement: https://github.com/Dasharo/open-source-firmware-validation/issues/82